### PR TITLE
Fix: search results selected language 

### PR DESCRIPTION
**Pull request by Bilelkihal**
# Fixed bugs:
- Add language params to search results links
  related issue: https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/594
  Demo:
  
[Capture vidéo du 23-04-2024 15:14:19.webm](https://github.com/ontoportal-lirmm/bioportal_web_ui/assets/61744974/e171bc87-aef3-4369-ae3b-b09e2c66a1b0)

- Remove duplicated ontologies notes (Already resolved https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/167)

- Make the selection of the closest preflabel to the search page query case unsensitive to fix:
"E2: Search “pimenta” in AgroPortal ... the first result is in arabic whereas the string pimenta exists in AGROVOC"
image

### CODE REVIEW

File: app/controllers/concerns/search_aggregator.rb [hunk 1]
```diff
@@ -24,39 +24,39 @@ module SearchAggregator
     /semanticweb/i
   ]
 
-  def aggregate_results(query, results)
+  def aggregate_results(query, results, lang)
```
Comment (syphax-bouazzouni) [inline app/controllers/concerns/search_aggregator.rb:27]: to not pass `lang` as parameter on each function, you can use the global helper `helpers.request_lang`

File: app/controllers/ontologies_controller.rb [hunk 2]
```diff
@@ -159,6 +159,7 @@ def new
 
   def notes
     @notes = @ontology.explore.notes
+    @notes = @notes.uniq { |note| [note.body, note.subject] }
```
Comment (syphax-bouazzouni) [inline app/controllers/ontologies_controller.rb:162]: this is not a solution, 
the issue is why the API return duplicate notes? do they have the same id ? if no, so they are not the same.

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/controllers/ontologies_controller.rb (hunk 2 above)
Comment (syphax-bouazzouni) [reply app/controllers/ontologies_controller.rb:162]: if we see the API, we see that the IDs are different
image

The issue was resolved here https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/167, a time ago. The bug was when we created a note, the event was triggered 4 times, creating multiple notes.

In summary to try to create note, and see it works well and create only one. And remove the current duplicate

Comment (Bilelkihal) [reply app/controllers/ontologies_controller.rb]: Yes u r right, the issue was in the creation, and now it's solved

**Review by syphax-bouazzouni — APPROVED** [review]

**Pull request merged**